README.md

```diff
@@ -28,4 +28,294 @@ One of the very common use cases of WebXR is 180 and/or 360 photos or videos, bo
+  - [XRCubeLayer](./cubemap.md)
+
+Layers are not presented to the XR device until they have been added to the `layers` `XRRenderState` property with the `updateRenderState()` method. Setting the `layers` array will override the `baseLayer` if one is present, with `baseLayer` reporting the first element in the `layers` array. Layers will be presented in the order they are given in the `layers` array, with each layers being given in "back-to-front" order. Layers may have alpha blending applied if the layer's `blendSourceAlpha` attribute is `true`, but no depth testing may be performed between layers.
+
+```js
+const projectionLayer = xrSesssion.requestLayer('projection');
+const quadLayer = xrSesssion.requestLayer('quad');
+
+xrSession.updateRenderState({ layers: [projectionLayer, quadLayer] });
+```
+
+`updateRenderState()` _may_ throw an exception if more layers are specified than the `XRSession` supports simultaneously.
+
+### Graphics API binding
+
+On their own, layers have no visual data, and will be ignored during compositing until provided with a backing `XRLayerSource`. A layer source is typically a GPU resource, like a texture, provided by one of the Web platform's graphics APIs. In order to specify which API is providing the layer sources an `XRGraphicsBinding` instance for the API in question must be created. For example, creating a graphics binding fow WebGL would function like this:
+
+```js
+const canvas = document.createElement('canvas');
+const gl = canvas.getContext('webgl', { xrCompatible: true });
+const xrGfx = new XRWebGLGraphicsBinding(xrSession, gl);
+```
+
+A session that wanted to take advantage of WebGL 2.0 techniques would alternatively create a `XRWebGL2GraphicsBinding` like so:
+
+```js
+const canvas = document.createElement('canvas');
+const gl = canvas.getContext('webgl2', { xrCompatible: true });
+const xrGfx = new XRWebGL2GraphicsBinding(xrSession, gl);
+```
+
+And a theoretical WebGPU graphics binding (which is not being proposed at this time and is offered only for illustrative purposes) may look like this:
+
+```js
+const gpuAdapter = await navigator.gpu.requestAdapter({ xrCompatible: true });
+const gpuDevice = await gpuAdapter.requestDevice();
+const xrGfx = new XRWebGPUGraphicsBinding(xrSession, gpuDevice);
+```
+
+Each graphics API may have unique requirements that must be satisfied before a context can be used in the creation of an `XRGraphicsBinding`. For example, a `WebGLRenderingContext` must have its `xrCompatible` flag set prior to being passed to the constructor of the `XRWebGLGraphicsBinding` instance.
+
+Any interaction between the `XRSession` the graphics API, such as allocating or retrieving textures, will go through this `XRGraphicsBinding` instance, and the exact mechanics of the interaction will typically be API specific. This allows the rest of the WebXR API to be graphics API agnostic and more easily adapt to future advances in rendering techniques.
+
+## Layer source allocation
```

Comment:
I still think that we should try to find a way to combine layer and graphics backing at the same time.
Otherwise we have to deal with all the edge case (ie what happens if a layer has no backing or gets backing later, what if multiple layer sources use the same layer)

Reply:
What is more interesting is the multiview case, where you render both views in one pass. We need a way to handle this.

Reply:
I am on opposite side, I really love separation of the graphics side from the layers themselves. Yes, the API becomes a bit more complex, but in long term it worth it, IMO.

Reply:
In C++ I would agree with you. 
However, this is javascript inside the browser. Usually interfaces are designed to *ONLY* do what is required for specific use cases. 
For instance, OpenXR allows any combination of layers, poses and swapchains regardless if they make sense. On the Web we should only allow constructs that are sensical.